### ISSUE TITLE
I wanna get more details about the interesting idea

### ISSUE THREAD

**Issue by hxpeckish** [WU-Dongze/MetaLoc#3]
Dear Professor,

I hope this message finds you well. I am an undergraduate student from China, about to graduate, and I am currently reading my first paper in this field. I must say, the paper is really well-written, and I am able to understand your ideas clearly. Moreover, I managed to reproduce part of the work based on the code you provided, which makes me very happy.

The reason I am reaching out is that your code does not include the MAML-TS algorithm part. I have made several attempts but have not been able to write the code that matches the task similarity in MAML-TS. If it’s not too much trouble, would you be able to share the MAML-TS portion of the code with me? I would be extremely grateful.

Thank you very much for your time and consideration.

**Comment by hxpeckish** [comment]
Hello Professor,
You may not have replied to my message because the algorithm I mentioned might already be included in the commented sections of the code. I will carefully read through those annotated parts next.
Also, I noticed that in the issue, you provided our csi.csv, csi_test.csv, and locationcomp.csi files, which are dated 0721. As far as I understand from your paper, this data was used for the NLOS environment, corresponding to the latter five days of collection. The earlier five days were collected in the hall, representing the LOS environment.
At the moment, I am lacking the corresponding test labels and location error data for the CSI.csv collected in the hall (i.e., LOS environment).
Could you please provide me with this part of the dataset?

**Comment by hpl20012373-lang** [comment]
> Dear Professor,  尊敬的教授：
> 
> I hope this message finds you well. I am an undergraduate student from China, about to graduate, and I am currently reading my first paper in this field. I must say, the paper is really well-written, and I am able to understand your ideas clearly. Moreover, I managed to reproduce part of the work based on the code you provided, which makes me very happy.我希望这条信息能让你一切顺利。我是一名来自中国的本科生，即将毕业，目前正在阅读我在这个领域的第一篇论文。我必须说，这篇论文写得真好，我能够清楚地理解你的想法。而且，我根据你提供的代码，成功地复制了部分工作，这让我非常高兴。
> 
> The reason I am reaching out is that your code does not include the MAML-TS algorithm part. I have made several attempts but have not been able to write the code that matches the task similarity in MAML-TS. If it’s not too much trouble, would you be able to share the MAML-TS portion of the code with me? I would be extremely grateful.我联系的原因是您的代码不包含 MAML-TS 算法部分。我已经进行了多次尝试，但无法编写与 MAML-TS 中的任务相似性相匹配的代码。如果不是太麻烦，你能和我分享代码的 MAML-TS 部分吗？我将非常感激。
> 
> Thank you very much for your time and consideration.非常感谢您的时间和考虑。

可以交流一下

**Comment by StatFusion** [comment]
Hi All,

Thanks for reaching out! We apologize for not responding to your previous message in time due to other commitments. Attached are the MAML-TS code files, including MMD.py, which calculates the Maximum Mean Discrepancy between the new testing environment’s data and each training environment’s data, and train_tsmaml.py, which is the training file.
 
The core idea of MAML-TS is to first individually train the neural network parameters $\theta_i$ for each training environment $i$. During testing, we employ MMD to identify the most similar training environment to the test, and we initialize the neural network with the parameters $\theta_i$ corresponding to that environment.
 
Hope that helps!
 
Best,
Dongze 

[MMD.py](https://github.com/user-attachments/files/22427927/MMD.py)
[train_tsmaml.py](https://github.com/user-attachments/files/22427928/train_tsmaml.py)

**Comment by hxpeckish** [comment]
Dear Professor,
  I am really surprised to receive your respond. For me, what you did is a
big hit that will make me have more power to explore relative questions.
Thank you very much!
  I'll make a deep learning for your code. By the way, I had finished the
code about MAML-TS by myself based on your detailed and great paper. I am
glad to tell you that I have entered the Wuhan university as a master,
majoring Intelligent processing of audio algorithms. maybe this interesting
paper will help me one day in the future.
  best~~~~~wishes for you!

  Best,
  Sanjin

StatFusion ***@***.***> 于2025年9月19日周五 21:10写道：
